proto/streamlit/proto/ButtonGroup.proto

```diff
@@ -0,0 +1,53 @@
+/**!
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+option java_package = "com.snowflake.apps.streamlit";
+option java_outer_classname = "ButtonGroupProto";
+
+message ButtonGroup {
+  enum ClickMode {
+      SINGLE_SELECT = 0;
+      MULTI_SELECT = 1;
+  }
+
+  message Option {
+    string content = 1;
+    string selected_content = 2;
+    // Can be used to disable the default selection highlight
+    // (in case the selected content is visualization enough)
+    bool disable_selection_highlight = 3;
```

Comment:
This flags seems a bit of unecessary overhead. Can we not just say activate this mode automatically if `selected_content` is provided?

Reply:
yeah good idea 👍 the original idea was that you might have `selected_content` but still want to disable highlighting in the future, but I guess its overkill for now.